Plutarch/DataRepr/Internal.hs

```diff
@@ -108,6 +107,12 @@ data PDataRecord (as :: [PLabeledType]) (s :: S) where
     PDataRecord ((name ':= x) ': xs) s
   PDNil :: PDataRecord '[] s
 
+instance {-# OVERLAPPABLE #-} PlutusType (PDataRecord l) where
+  type PInner (PDataRecord l) _ = PBuiltinList PData
+  pcon' (PDCons x xs) = pcons # pforgetData x # pto xs
+  pcon' PDNil         = pnil
```

Comment:
these should probably also err

Reply:
or maybe not, doesn't seem like they are any less efficient? anyway, I think you can actually do `pcon' (PDCons x xs) = pcon' (PDCons x xs)` and have it work

Reply:
You should probably add type annotations to make it clear what's happening

Reply:
added type annotations, also made it use the underlying `pcon'` instead - good idea!